src/OrleansServiceBus/Providers/Streams/EventHub/EventHubAdapterFactory.cs

```diff
@@ -283,6 +283,12 @@ private EventHubAdapterReceiver GetOrCreateReceiver(QueueId queueId)
             return cache;
         }
 
+        protected virtual IEventHubQueueCache CreateCache(IStreamQueueCheckpointer<string> checkpointer, 
```

Comment:
That sounds like a good idea.

We have an abstraction for that, IQueueAdapterCache, but I'm not really sure it's a good fit.  Due to the tighter coupling of the receiver and the cache (in this stream provider), we've not really used it as intended.

We also have a factory pattern in DI now (see src\Orleans\Utils\Factory.cs).  The issue with using DI in these cases has been the fact that we can't assume that every stream provider wants the same caching patterns.  We do have a DI disambiguation pattern (see IKeyedServiceCollection) now, which can be used to get a cache factory for a specific stream provider, but that's not really been thought through yet.